File: Logica_de_Programacao_Python/Desafio_2/desafio_2/Desafio2.py
def calcular_desconto(quantidade, preco):
    if quantidade <= 10:
        desconto = 0  # NÃO POSSUI DESCONTO
    elif 11 <= quantidade <= 20:
        desconto = 0.10  # 10% DE DESCONTO
    elif 21 <= quantidade <= 50:
        desconto = 0.20  # 20% DE DESCONTO
    else:
        desconto = 0.25  # 25% DE DESCONTO

    subtotal = quantidade * preco
    valor_com_desconto = subtotal * (1 - desconto)
    return desconto, valor_com_desconto


def main():
    print("Bem-vindo ao sistema de compras do mercado!\n")
    produtos = []

    while True:
        nome = input("Informe o nome do produto (ou digite 'sair' para finalizar): ")
        if nome.lower() == "sair":
            break

        preco = float(input(f"Informe o preço do produto '{nome}': R$ "))
        quantidade = int(input(f"Informe a quantidade de '{nome}': "))

        produtos.append({"nome": nome, "preco": preco, "quantidade": quantidade})
        print("\n")

    # EXIBINDO OS DETALHES DOS PRODUTOS E CALCULANDO O TOTAL GERAL
    print("\nResumo da Compra:")
    valor_total_geral = 0

    for produto in produtos:
        desconto, valor_com_desconto = calcular_desconto(
            produto["quantidade"], produto["preco"]
        )
        valor_total_geral += valor_com_desconto

        # EXIBE O SUBTOTAL
        detalhes = (
            f"- {produto['nome']}: {produto['quantidade']} unidade(s) x R$ {produto['preco']:.2f} "
            f"= R$ {produto['quantidade'] * produto['preco']:.2f}"
        )
        # INCLUI O DESCONTO SOMENTE SE APLICÁVEL
        if desconto > 0:
            detalhes += f" (Desconto: {desconto * 100:.0f}%, Total com desconto: R$ {valor_com_desconto:.2f})"
        print(detalhes)

    # EXIBINDO O TOTAL GERAL
    print(f"\nValor total a pagar: R$ {valor_total_geral:.2f}")

if __name__ == "__main__":
    main()

File: Logica_De_Programacao_Java/Desafio_2/src/Desafio_2.java
import java.util.Scanner;

public class Desafio_2 {
    public static void main(String[] args) {
        Scanner scanner = new Scanner(System.in);

        System.out.println("Digite o nome do produto:");
        String nomeProduto = scanner.nextLine();

        System.out.println("Digite o preço unitário do produto:");
        double preco = scanner.nextDouble();

        System.out.println("Digite a quantidade comprada:");
        int quantidade = scanner.nextInt();

        double valorTotal = preco * quantidade;
        double desconto = 0;

        if (quantidade > 50) {
            desconto = 0.25;
        } else if (quantidade >= 21) {
            desconto = 0.20;
        } else if (quantidade >= 11) {
            desconto = 0.10;
        }

        double valorComDesconto = valorTotal - (valorTotal * desconto);

        System.out.println("\nResumo da Compra:");
        System.out.println("Produto: " + nomeProduto);
        System.out.println("Quantidade: " + quantidade);
        System.out.printf("Valor Total: R$ %.2f\n", valorTotal);
        System.out.printf("Desconto aplicado: %.0f%%\n", desconto * 100);
        System.out.printf("Valor Final a Pagar: R$ %.2f\n", valorComDesconto);

        scanner.close();
    }
}